Assets/Scripts/DestroyByContact.cs

```diff
@@ -14,21 +17,25 @@ public class DestroyByContact : MonoBehaviour
 
     void OnTriggerEnter(Collider other)
     {
-        if (other.tag == "Boundary")
+        if (other.tag.Equals(BoundaryTag))
         {
             return;
         }
 
         Instantiate(explosion, transform.position, transform.rotation);
 
-        if (other.tag == "Player")
+        if (other.tag.Equals(PlayerTag))
         {
             Instantiate(playerExplosion, other.transform.position, other.transform.rotation);
-			GameController.instance.GameOver();
+            GameController.Instance.GameOver();
         }
         else
-            GameController.instance.AddScore(scoreValue);
-        Destroy(other.gameObject);
+            GameController.Instance.AddScore(scoreValue);
+
+        if (!(other.tag.Equals(BombTag)))
```

Comment:
why do you have the extra parentheses? 

```c#
if(!other.tag.Equals(BombTag, StringComparison.Ordinal)) 
{
     // stuff
}

// or

if(other.tag.Equals(BombTag) == false)
{
    // do stuff
}
```

But don't do the extra parentheses, they just waste time of the programmer reading the code as they try to figure out why you scoped it like this